src/coreComponents/common/TypeDispatch.hpp

```diff
@@ -254,30 +300,38 @@ bool dispatchViaTable( TypeList< Ts... > const types,
 
 /**
  * @brief Dispatch a generic worker function @p lambda based on runtime type.
- * @tparam Ts list of types
- * @tparam LAMBDA type of user-provided function or lambda, must have one auto argument
- * @param types list of types as an object (for deduction)
- * @param type type index of the runtime type
- * @param errorIfTypeNotFound flag indicating whether dispatch should issue an error when type not matched
- * @param lambda user-provided callable, will be called with a single prvalue of type indicated by @p type
+ *
+ * @tparam LIST type of the list of types
+ * @tparam LAMBDA  type of user-provided function or lambda, must have one auto argument
+ * @tparam Ts types of the objects to be dispatched.
+ * @param combinations list of types
+ * @param lambda lambda user-provided callable
+ * @param objects objects to be dispatched
  * @return @p true iff type has been dispatch
  *
- * @todo Do we want @p errorIfTypeNotFound parameter? Options:
+ * todo Do we want errorIfTypeNotFound parameter? Options:
  *       - make it a template parameter (caller always knows whether or not it wants hard errors)
  *       - make the caller process return value and raise error if needed (and however they want)
  */
-template< typename ... Ts, typename LAMBDA >
-bool dispatch( TypeList< Ts... > const types,
-               std::type_index const type,
-               bool const errorIfTypeNotFound,
-               LAMBDA && lambda )
+template< typename LIST, typename LAMBDA, typename ... Ts >
+bool dispatch( LIST const combinations,
+               LAMBDA && lambda,
+               Ts && ... objects )
 {
-  bool const success = internal::dispatchViaTable( types, type, std::forward< LAMBDA >( lambda ) );
-  if( !success && errorIfTypeNotFound )
+  bool const success = internal::dispatchViaTable( combinations,
+                                                   std::forward< LAMBDA >( lambda ),
+//                                                   std::make_tuple( std::type_index( internal::typeIdWrapper(objects))... ) );
+                                                   std::make_tuple( std::type_index( typeid(objects))... ) );
+
+  if( !success )
   {
-    GEOS_ERROR( "Type " << LvArray::system::demangle( type.name() ) << " was not dispatched.\n" <<
-                "Check the stack trace below and revise the type list passed to dispatch().\n" <<
-                "If you are unsure about this error, please report it to GEOSX issue tracker." );
+    std::ostringstream dispatchInput;
+    ( (dispatchInput<<"\n  "<<LvArray::system::demangle( internal::typeIdWrapper( objects ).name() ) ), ...);
```

Comment:
@klevzoff can this be done without the `ostringstream`?

Reply:
IIRC something like:

`(... + ( "  \n" + LvArray::system::demangle( internal::typeIdWrapper( objects ).name() ) ) ))`

should work, though I think that will construct many temporary strings.

Reply:
(assuming `.name()` is returning a string and on a character pointer